File: SmartTooling-SPA/src/app/_core/_models/best-line/bl-layout-design-overall.ts
import { BL_Lines } from "./bl-lines";
import { BLAttachments } from "./bl_attachments";
import { BL_Critical_Process_Analysis } from "./bl_critical_process_analysis_audit";
import { BL_Layout_Design_Process } from "./bL_Layout_Design_Process";

export interface BLLayoutDesignOverall {
  id: number;
  factory_id: string;
  line_id: string;
  line_type_id: string;
  model_no: string;
  prod_season: string;
  no_of_process_before: number | null;
  no_of_process_after: number | null;
  tct_before: number;
  tct_after: number;
  cps_mp_before: number;
  cps_mp_after: number;
  assembly_mp_before: number;
  assembly_mp_after: number;
  eolr_before: number;
  eolr_after: number;
  ller_before_percent: number;
  ller_after_percent: number;
  tentative_pph_before: number;
  tentative_pph_after: number;
  tentative_efficiency_before_percent: number | null;
  tentative_efficiency_after_percent: number | null;
  c2b_overall_image: string;
  create_by: string;
  create_time: string;
  update_by: string;
  update_time: string;
  line_: BL_Lines;
  bL_Attachments: BLAttachments[];
  bL_Critical_Process_Analysis: BL_Critical_Process_Analysis[];
  bL_Layout_Design_Process: BL_Layout_Design_Process[];
}

File: SmartTooling-API/Models/BestLine/BL_Layout_Design_Overall.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartTooling_API.Models.BestLine
{
    public partial class BL_Layout_Design_Overall
    {
        public BL_Layout_Design_Overall()
        {
            BL_Attachments = new HashSet<BL_Attachments>();
            BL_Critical_Process_Analysis = new HashSet<BL_Critical_Process_Analysis>();
            BL_Layout_Design_Process = new HashSet<BL_Layout_Design_Process>();
        }

        [Key]
        public long id { get; set; }
        [Required]
        [StringLength(50)]
        public string factory_id { get; set; }
        [Required]
        [StringLength(10)]
        public string line_id { get; set; }
        [Required]
        [StringLength(20)]
        public string line_type_id { get; set; }
        [Required]
        [StringLength(8)]
        public string model_no { get; set; }
        [Required]
        [StringLength(4)]
        public string prod_season { get; set; }
        [Column(TypeName = "numeric(18, 0)")]
        public decimal? no_of_process_before { get; set; }
        [Column(TypeName = "numeric(18, 0)")]
        public decimal? no_of_process_after { get; set; }
        [Column(TypeName = "numeric(18, 0)")]
        public decimal tct_before { get; set; }
        [Column(TypeName = "numeric(18, 0)")]
        public decimal tct_after { get; set; }
        [Column(TypeName = "numeric(18, 0)")]
        public decimal cps_mp_before { get; set; }
        [Column(TypeName = "numeric(18, 0)")]
        public decimal cps_mp_after { get; set; }
        [Column(TypeName = "numeric(18, 0)")]
        public decimal assembly_mp_before { get; set; }
        [Column(TypeName = "numeric(18, 0)")]
        public decimal assembly_mp_after { get; set; }
        [Column(TypeName = "numeric(18, 0)")]
        public decimal eolr_before { get; set; }
        [Column(TypeName = "numeric(18, 0)")]
        public decimal eolr_after { get; set; }
        [Column(TypeName = "numeric(18, 2)")]
        public decimal ller_before_percent { get; set; }
        [Column(TypeName = "numeric(18, 2)")]
        public decimal ller_after_percent { get; set; }
        [Column(TypeName = "numeric(18, 2)")]
        public decimal tentative_pph_before { get; set; }
        [Column(TypeName = "numeric(18, 2)")]
        public decimal tentative_pph_after { get; set; }
        [Column(TypeName = "numeric(18, 2)")]
        public decimal? tentative_efficiency_before_percent { get; set; }
        [Column(TypeName = "numeric(18, 2)")]
        public decimal? tentative_efficiency_after_percent { get; set; }
        [StringLength(200)]
        public string c2b_overall_image { get; set; }
        [Required]
        [StringLength(50)]
        public string create_by { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime create_time { get; set; }
        [Required]
        [StringLength(50)]
        public string update_by { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime update_time { get; set; }

        public virtual BL_Lines line_ { get; set; }
        [InverseProperty("layout_design_overall_")]
        public virtual ICollection<BL_Attachments> BL_Attachments { get; set; }
        [InverseProperty("layout_design_overall_")]
        public virtual ICollection<BL_Critical_Process_Analysis> BL_Critical_Process_Analysis { get; set; }
        [InverseProperty("layout_design_overall_")]
        public virtual ICollection<BL_Layout_Design_Process> BL_Layout_Design_Process { get; set; }
    }
}
